pwndbg/gdblib/onegadget.py

```diff
@@ -0,0 +1,595 @@
+from __future__ import annotations
+
+import ctypes
+import hashlib
+import os
+import re
+import subprocess
+from enum import Enum
+
+import gdb
+from tabulate import tabulate
+
+import pwndbg.color.message as M
+import pwndbg.gdblib.arch
+import pwndbg.gdblib.file
+import pwndbg.gdblib.memory
+import pwndbg.gdblib.vmmap
+import pwndbg.glibc
+import pwndbg.lib.cache
+import pwndbg.lib.tempfile
+from pwndbg.color import colorize
+from pwndbg.color import generateColorFunction
+
+ADDRESS_WRITABLE_PATTERN = re.compile(r"address(?:es)? (.*) (?:is|are) writable")
+WRITABLE_COLON_PATTERN = re.compile(r"writable: (.*)")
+EQUAL_NULL_PATTERN = re.compile(r"(.*) == NULL")
+VALID_POSIX_SPAWN_FILE_ACTIONS_PATTERN = re.compile(r"(.*) <= 0")
+VALID_ARGV_PATTERN = re.compile(r"(.+) is a valid argv")
+VALID_ENVP_PATTERN = re.compile(r"(.+) is a valid envp")
+IS_ALIGNED_PATTERN = re.compile(r"(.+) & 0xf == (\d+)")
+IS_GOT_ADDRESS_PATTERN = re.compile(r"(.+) is the GOT address of libc")
+CAST_PATTERN = re.compile(r"^\([s|u]\d+\)")
```

Comment:
Fwiw all this `re.compile` etc. will slow down Pwndbg load time, but it seems negligible:
```
pwndbg> !cat a.py
import re
import time

a = time.time()
ADDRESS_WRITABLE_PATTERN = re.compile(r"address(?:es)? (.*) (?:is|are) writable")
WRITABLE_COLON_PATTERN = re.compile(r"writable: (.*)")
EQUAL_NULL_PATTERN = re.compile(r"(.*) == NULL")
VALID_POSIX_SPAWN_FILE_ACTIONS_PATTERN = re.compile(r"(.*) <= 0")
VALID_ARGV_PATTERN = re.compile(r"(.+) is a valid argv")
VALID_ENVP_PATTERN = re.compile(r"(.+) is a valid envp")
IS_ALIGNED_PATTERN = re.compile(r"(.+) & 0xf == (\d+)")
IS_GOT_ADDRESS_PATTERN = re.compile(r"(.+) is the GOT address of libc")
CAST_PATTERN = re.compile(r"^\([s|u]\d+\)")
b = time.time()

elapsed = b - a
print(elapsed)
pwndbg> source a.py
0.0009448528289794922
```